src/plugins/intel_npu/src/plugin/npuw/just_sync_infer_request.cpp

```diff
@@ -530,6 +530,81 @@ ov::npuw::IBaseInferRequest::RqPtr ov::npuw::JustInferRequest::get_real_subreque
     return m_subrequests[real_idx];
 }
 
+void ov::npuw::JustInferRequest::log_subrequest_launch(std::size_t idx, std::size_t real_idx) {
+    LOG_INFO("Launching subrequest[" << idx << "]" <<
+        ((real_idx == idx) ? std::string("...").c_str() :
+                             std::string(std::string(", which is actually subrequest[") +
+                                 std::to_string(real_idx) + "]...").c_str())); 
+}
+
+void ov::npuw::JustInferRequest::log_launch_on_range(std::size_t real_idx, std::size_t dim,
+                                                     std::size_t offset, std::size_t len) {
+    LOG_INFO("Launch subrequest[" << real_idx << "] on range for dim=" <<
+        dim << " : [" << offset << ", " << offset + len << ")");
+}
+
+void ov::npuw::JustInferRequest::try_accurate_infer(std::size_t real_idx, bool& failover) {
+    NPUW_ASSERT(m_npuw_model->m_compiled_submodels.at(real_idx).replaced_by.value_or(real_idx) == real_idx);
+
+    m_subrequests[real_idx]->infer();
+    if (m_npuw_model->m_acc_check) {
+        ensure_subrequest_is_accurate(real_idx, failover);
+    }
+
+}
+
+void ov::npuw::JustInferRequest::ensure_subrequest_is_accurate(std::size_t idx, bool& failover) {
+    LOG_INFO("Check if subrequest[" << idx << "] is accurate...");
+    LOG_BLOCK();
+    failover = false;
+    if (m_ref_subrequests.at(idx) != nullptr && m_subrequests.at(idx)._ptr != m_ref_subrequests.at(idx)._ptr) {
+        NPUW_ASSERT(m_npuw_model->m_compiled_submodels.at(idx).switched_to_ref == false);
+        NPUW_ASSERT(m_npuw_model->m_compiled_submodels.at(idx).replaced_by.value_or(idx) == idx);
+
+        const auto& ref_comp_model = m_ref_subrequests.at(idx)->get_compiled_model();
+        const auto& actual_comp_model = m_subrequests.at(idx)->get_compiled_model();
+        NPUW_ASSERT(actual_comp_model->inputs().size() == ref_comp_model->inputs().size());
+        // Setting inputs:
+        for (size_t i = 0; i < actual_comp_model->inputs().size(); i++) {
+            const auto& itensor = m_subrequests.at(idx)->get_tensor(actual_comp_model->inputs()[i]);
+            m_ref_subrequests.at(idx)->set_tensor(ref_comp_model->inputs()[i], itensor);
```

Comment:
I'd recommend to just take references on the `actual` and `reference` requests once and don't do `.at()`ing much

Reply:
Fixed, thanks!